src/core/Root.ts

```diff
@@ -1,14 +1,85 @@
-import { Diagram } from "../diagram/Diagram";
-// import { MainApp } from "./MainApp"; // To be continued...
+import { CancellableCommand } from "../command/Commands";
+import { Diagram, Memento } from "../diagram/Diagram";
+import { Timeline } from "../diagram/Timeline";
+import { MainApp } from "./MainApp";
 import { Modals } from "./Modals";
 
 export type RootStore = typeof rootStore;
 
 const rootStore = {
-  diagram: new Diagram(),
+  app: new MainApp(),
   modals: new Modals()
 } as const;
 
 export function getRootStore(): RootStore {
   return rootStore;
 }
+
+export class MainDiagramHandler extends Timeline<CancellableCommand> {
+  // The current memento saved in the file or the first memento in the timeline
+  private sourceCurrentMemento!: Memento | null;
+  private isModifiedbool!: boolean;
```

Comment:
Change `isModifiedbool` to `isModifiedFlag`

Reply:
Updated.